KotlinParseException: Your text is not valid

The exception occurs on the project okhttp

[ERROR] 2021-06-09 16:15:57 Internal error has occurred in class-compact-initialization. Please make an issue on this bug at https://github.com/cqfn/diKTat/.
org.cqfn.diktat.ruleset.utils.KotlinParseException: Your text is not valid
        at org.cqfn.diktat.ruleset.utils.KotlinParser.createNode(KotlinParser.kt:75)
        at org.cqfn.diktat.ruleset.utils.KotlinParser.createNode$default(KotlinParser.kt:75)
        at org.cqfn.diktat.ruleset.rules.chapter6.classes.CompactInitialization.getOrCreateApplyBlock(CompactInitialization.kt:124)
        at org.cqfn.diktat.ruleset.rules.chapter6.classes.CompactInitialization.moveAssignmentIntoApply(CompactInitialization.kt:80)
        at org.cqfn.diktat.ruleset.rules.chapter6.classes.CompactInitialization.access$moveAssignmentIntoApply(CompactInitialization.kt:30)
        at org.cqfn.diktat.ruleset.rules.chapter6.classes.CompactInitialization$handleProperty$2$1.invoke(CompactInitialization.kt:72)
        at org.cqfn.diktat.ruleset.rules.chapter6.classes.CompactInitialization$handleProperty$2$1.invoke(CompactInitialization.kt:71)
        at org.cqfn.diktat.ruleset.constants.Warnings.warnAndFix(Warnings.kt:259)
        at org.cqfn.diktat.ruleset.constants.Warnings.warnAndFix$default(Warnings.kt:201)

--- a/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
+++ b/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
@@ -278,22 +278,23 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
         node.replaceChild(whiteSpaceAfterPlus, PsiWhiteSpaceImpl("\n"))
     }
 
+    @Suppress("UnsafeCallOnNullableType")
     private fun fixStringTemplate(wrongStringTemplate: LongLineFixableCases.StringTemplate) {
         val incorrectText = wrongStringTemplate.node.text
         val firstPart = incorrectText.substring(0, wrongStringTemplate.delimiterIndex)
         val secondPart = incorrectText.substring(wrongStringTemplate.delimiterIndex, incorrectText.length)
         val isSplitInWhiteSpace = wrongStringTemplate.node.psi.findElementAt(wrongStringTemplate.delimiterIndex)!!.node.isWhiteSpace()
         val correctNode =
-            if (!isSplitInWhiteSpace) {
-                KotlinParser().createNode("$firstPart\" +\n\"$secondPart")
-            } else {
-                // case when split space end in string interpolation
-                // "Hello ${if (true) "Alice" else "Nick"}"
-                //                                ^
-                //                                |
-                //                           split space
-                KotlinParser().createNode("$firstPart \n$secondPart")
-            }
+                if (!isSplitInWhiteSpace) {
+                    KotlinParser().createNode("$firstPart\" +\n\"$secondPart")
+                } else {
+                    // case when split space end in string interpolation
+                    // "Hello ${if (true) "Alice" else "Nick"}"
+                    //                                ^
+                    //                                |
+                    //                           split space
+                    KotlinParser().createNode("$firstPart \n$secondPart")
+                }
         wrongStringTemplate.node.treeParent.replaceChild(wrongStringTemplate.node, correctNode)
     }
 

--- a/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
+++ b/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
@@ -143,8 +143,18 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
      *         None - if the string can't be split
      */
     private fun checkStringTemplate(node: ASTNode, configuration: LineLengthConfiguration): LongLineFixableCases {
-        val leftOffset = positionByOffset(node.startOffset).second
-        val delimiterIndex = node.text.substring(0, configuration.lineLength.toInt() - leftOffset).lastIndexOf(' ')
+        var multiLineOffset = 0
+        val (nodeText, leftOffset) = if (node.text.lines().size > 1 ) {
+            node
+                .text
+                .lines()
+                .takeWhile { it.length < configuration.lineLength }
+                .forEach { multiLineOffset += it.length }
+            node.text.lines().find { it.length > 120 }!! to positionByOffset(node.psi.findElementAt(multiLineOffset + 1)!!.node.startOffset).second
+        } else {
+            node.text to positionByOffset(node.startOffset).second
+        }
+        val delimiterIndex = nodeText.substring(0, configuration.lineLength.toInt() - leftOffset).lastIndexOf(' ')
         if (delimiterIndex == -1) {
             // we can't split this string, however may be we can move it entirely:
             // case when new line should be inserted after `+`. Example: "first" + "second"
@@ -156,11 +166,11 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
         }
         // minus 2 here as we are inserting ` +` and we don't want it to exceed line length
         val correcterDelimiter = if (leftOffset + delimiterIndex > configuration.lineLength.toInt() - 2) {
-            node.text.substring(0, delimiterIndex - 2).lastIndexOf(' ')
+            nodeText.substring(0, delimiterIndex - 2).lastIndexOf(' ')
         } else {
             delimiterIndex
         }
-        return LongLineFixableCases.StringTemplate(node, correcterDelimiter)
+        return LongLineFixableCases.StringTemplate(node, correcterDelimiter, multiLineOffset)
     }
 
     private fun checkFun(wrongNode: ASTNode) =
@@ -283,7 +293,7 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
         val incorrectText = wrongStringTemplate.node.text
         val firstPart = incorrectText.substring(0, wrongStringTemplate.delimiterIndex)
         val secondPart = incorrectText.substring(wrongStringTemplate.delimiterIndex, incorrectText.length)
-        val isSplitInWhiteSpace = wrongStringTemplate.node.psi.findElementAt(wrongStringTemplate.delimiterIndex)!!.node.isWhiteSpace()
+        val isSplitInWhiteSpace = wrongStringTemplate.multiLineOffset == 0 && wrongStringTemplate.node.psi.findElementAt(wrongStringTemplate.delimiterIndex)!!.node.isWhiteSpace()
         val correctNode =
                 if (!isSplitInWhiteSpace) {
                     KotlinParser().createNode("$firstPart\" +\n\"$secondPart")
@@ -470,7 +480,7 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
             val hasNewLineBefore: Boolean,
             val indexLastSpace: Int = 0) : LongLineFixableCases()
 
-        class StringTemplate(val node: ASTNode, val delimiterIndex: Int) : LongLineFixableCases()
+        class StringTemplate(val node: ASTNode, val delimiterIndex: Int, val multiLineOffset: Int) : LongLineFixableCases()
 
         class BinaryExpression(val node: ASTNode) : LongLineFixableCases()
 

--- a/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
+++ b/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
@@ -142,6 +142,7 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
      *         BinaryExpression - if there is two concatenated strings and new line should be inserted after `+`
      *         None - if the string can't be split
      */
+    @Suppress("UnsafeCallOnNullableType")
     private fun checkStringTemplate(node: ASTNode, configuration: LineLengthConfiguration): LongLineFixableCases {
         var multiLineOffset = 0
         val (nodeText, leftOffset) = if (node.text.lines().size > 1 ) {
@@ -150,7 +151,7 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
                 .lines()
                 .takeWhile { it.length < configuration.lineLength }
                 .forEach { multiLineOffset += it.length }
-            node.text.lines().find { it.length > 120 }!! to positionByOffset(node.psi.findElementAt(multiLineOffset + 1)!!.node.startOffset).second
+            node.text.lines().find { it.length > configuration.lineLength }!! to positionByOffset(node.psi.findElementAt(multiLineOffset + 1)!!.node.startOffset).second
         } else {
             node.text to positionByOffset(node.startOffset).second
         }
@@ -293,6 +294,7 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
         val incorrectText = wrongStringTemplate.node.text
         val firstPart = incorrectText.substring(0, wrongStringTemplate.delimiterIndex)
         val secondPart = incorrectText.substring(wrongStringTemplate.delimiterIndex, incorrectText.length)
+        // wrongStringTemplate.multiLineOffset equals zero if string in one line
         val isSplitInWhiteSpace = wrongStringTemplate.multiLineOffset == 0 && wrongStringTemplate.node.psi.findElementAt(wrongStringTemplate.delimiterIndex)!!.node.isWhiteSpace()
         val correctNode =
                 if (!isSplitInWhiteSpace) {

--- a/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
+++ b/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
@@ -171,7 +171,8 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
             return LongLineFixableCases.None
         }
         // check, that space to split is a part of text - not code
-        if (node.psi.findElementAt(delimiterIndex)!!.node.isWhiteSpace()) {
+        val isSpaceIsWhiteSpace = node.psi.findElementAt(delimiterIndex)!!.node.isWhiteSpace()
+        if (isSpaceIsWhiteSpace) {
             return LongLineFixableCases.None
         }
         // minus 2 here as we are inserting ` +` and we don't want it to exceed line length

--- a/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
+++ b/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
@@ -142,7 +142,7 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
      *         BinaryExpression - if there is two concatenated strings and new line should be inserted after `+`
      *         None - if the string can't be split
      */
-    @Suppress("UnsafeCallOnNullableType")
+    @Suppress("UnsafeCallOnNullableType", "TOO_LONG_FUNCTION")
     private fun checkStringTemplate(node: ASTNode, configuration: LineLengthConfiguration): LongLineFixableCases {
         var multiLineOffset = 0
         val leftOffset = if (node.text.lines().size > 1) {

--- a/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
+++ b/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
@@ -154,7 +154,7 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
             node
                 .text
                 .lines()
-                .find { it.length > configuration.lineLength }!!
+                .first { it.length > configuration.lineLength }
                 .takeWhile { it.isWhitespace() }
                 .count()
         } else {
@@ -171,6 +171,7 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
             return LongLineFixableCases.None
         }
         // check, that space to split is a part of text - not code
+        // If the space split is part of the code, then there is a chance of breaking the code when fixing, that why we should ignore it
         val isSpaceIsWhiteSpace = node.psi.findElementAt(delimiterIndex)!!.node.isWhiteSpace()
         if (isSpaceIsWhiteSpace) {
             return LongLineFixableCases.None

--- a/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
+++ b/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
@@ -6,7 +6,6 @@ import org.cqfn.diktat.common.config.rules.getRuleConfig
 import org.cqfn.diktat.ruleset.constants.Warnings.LONG_LINE
 import org.cqfn.diktat.ruleset.rules.DiktatRule
 import org.cqfn.diktat.ruleset.utils.KotlinParser
-import org.cqfn.diktat.ruleset.utils.KotlinParseException
 import org.cqfn.diktat.ruleset.utils.appendNewlineMergingWhiteSpace
 import org.cqfn.diktat.ruleset.utils.calculateLineColByOffset
 import org.cqfn.diktat.ruleset.utils.findParentNodeWithSpecificType
@@ -283,20 +282,17 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
         val incorrectText = wrongStringTemplate.node.text
         val firstPart = incorrectText.substring(0, wrongStringTemplate.delimiterIndex)
         val secondPart = incorrectText.substring(wrongStringTemplate.delimiterIndex, incorrectText.length)
+        val isSplitInWhiteSpace = wrongStringTemplate.node.psi.findElementAt(wrongStringTemplate.delimiterIndex)!!.node.isWhiteSpace()
         val correctNode =
-            try {
+            if (!isSplitInWhiteSpace) {
                 KotlinParser().createNode("$firstPart\" +\n\"$secondPart")
-            } catch (ex: KotlinParseException) {
-                try {
-                    // case when split space end in string interpolation
-                    // "Hello ${if (true) "Alice" else "Nick"}"
-                    //                                ^
-                    //                                |
-                    //                           split space
-                    KotlinParser().createNode("$firstPart \n$secondPart")
-                } catch (ex: KotlinParseException) {
-                    return
-                }
+            } else {
+                // case when split space end in string interpolation
+                // "Hello ${if (true) "Alice" else "Nick"}"
+                //                                ^
+                //                                |
+                //                           split space
+                KotlinParser().createNode("$firstPart \n$secondPart")
             }
         wrongStringTemplate.node.treeParent.replaceChild(wrongStringTemplate.node, correctNode)
     }

--- a/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
+++ b/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
@@ -145,7 +145,7 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
     @Suppress("UnsafeCallOnNullableType")
     private fun checkStringTemplate(node: ASTNode, configuration: LineLengthConfiguration): LongLineFixableCases {
         var multiLineOffset = 0
-        val leftOffset = if (node.text.lines().size > 1 ) {
+        val leftOffset = if (node.text.lines().size > 1) {
             node
                 .text
                 .lines()
@@ -171,10 +171,12 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
             return LongLineFixableCases.None
         }
         // check, that space to split is a part of text - not code
-        if (node.psi.findElementAt(delimiterIndex)!!.node.isWhiteSpace())
+        if (node.psi.findElementAt(delimiterIndex)!!.node.isWhiteSpace()) {
             return LongLineFixableCases.None
+        }
         // minus 2 here as we are inserting ` +` and we don't want it to exceed line length
-        val correcterDelimiter = if (multiLineOffset ==0 && leftOffset + delimiterIndex > configuration.lineLength.toInt() - 2) {
+        val shouldAddTwoSpaces = multiLineOffset == 0 && leftOffset + delimiterIndex > configuration.lineLength.toInt() - 2
+        val correcterDelimiter = if (shouldAddTwoSpaces) {
             node.text.substring(0, delimiterIndex - 2).lastIndexOf(' ')
         } else {
             delimiterIndex
@@ -303,11 +305,11 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
         val firstPart = incorrectText.substring(0, wrongStringTemplate.delimiterIndex)
         val secondPart = incorrectText.substring(wrongStringTemplate.delimiterIndex, incorrectText.length)
         val textBetwenParts =
-            if (wrongStringTemplate.isOneLineString) {
-                "\" +\n\""
-            } else {
-                "\n"
-            }
+                if (wrongStringTemplate.isOneLineString) {
+                    "\" +\n\""
+                } else {
+                    "\n"
+                }
         val correctNode = KotlinParser().createNode("$firstPart$textBetwenParts$secondPart")
         wrongStringTemplate.node.treeParent.replaceChild(wrongStringTemplate.node, correctNode)
     }
@@ -487,9 +489,12 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
         /**
          * @property node node
          * @property delimiterIndex index to split
-         * @property isMultiString flag is string is multiline
+         * @property isOneLineString flag is string is one line
          */
-        class StringTemplate(val node: ASTNode, val delimiterIndex: Int, val isOneLineString: Boolean) : LongLineFixableCases()
+        class StringTemplate(
+            val node: ASTNode,
+            val delimiterIndex: Int,
+            val isOneLineString: Boolean) : LongLineFixableCases()
 
         class BinaryExpression(val node: ASTNode) : LongLineFixableCases()
 

--- a/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
+++ b/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
@@ -145,17 +145,22 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
     @Suppress("UnsafeCallOnNullableType")
     private fun checkStringTemplate(node: ASTNode, configuration: LineLengthConfiguration): LongLineFixableCases {
         var multiLineOffset = 0
-        val (nodeText, leftOffset) = if (node.text.lines().size > 1 ) {
+        val leftOffset = if (node.text.lines().size > 1 ) {
             node
                 .text
                 .lines()
                 .takeWhile { it.length < configuration.lineLength }
                 .forEach { multiLineOffset += it.length }
-            node.text.lines().find { it.length > configuration.lineLength }!! to positionByOffset(node.psi.findElementAt(multiLineOffset + 1)!!.node.startOffset).second
+            node
+                .text
+                .lines()
+                .find { it.length > configuration.lineLength }!!
+                .takeWhile { it.isWhitespace() }
+                .count()
         } else {
-            node.text to positionByOffset(node.startOffset).second
+            positionByOffset(node.startOffset).second
         }
-        val delimiterIndex = nodeText.substring(0, configuration.lineLength.toInt() - leftOffset).lastIndexOf(' ')
+        val delimiterIndex = node.text.substring(0, multiLineOffset + configuration.lineLength.toInt() - leftOffset).lastIndexOf(' ')
         if (delimiterIndex == -1) {
             // we can't split this string, however may be we can move it entirely:
             // case when new line should be inserted after `+`. Example: "first" + "second"
@@ -165,13 +170,10 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
             // can't fix this case
             return LongLineFixableCases.None
         }
-        // minus 2 here as we are inserting ` +` and we don't want it to exceed line length
-        val correcterDelimiter = if (leftOffset + delimiterIndex > configuration.lineLength.toInt() - 2) {
-            nodeText.substring(0, delimiterIndex - 2).lastIndexOf(' ')
-        } else {
-            delimiterIndex
-        }
-        return LongLineFixableCases.StringTemplate(node, correcterDelimiter, multiLineOffset)
+        // check, that space to split is a part of text - not code
+        if (node.psi.findElementAt(delimiterIndex)!!.node.isWhiteSpace())
+            return LongLineFixableCases.None
+        return LongLineFixableCases.StringTemplate(node, delimiterIndex, multiLineOffset == 0)
     }
 
     private fun checkFun(wrongNode: ASTNode) =
@@ -294,20 +296,13 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
         val incorrectText = wrongStringTemplate.node.text
         val firstPart = incorrectText.substring(0, wrongStringTemplate.delimiterIndex)
         val secondPart = incorrectText.substring(wrongStringTemplate.delimiterIndex, incorrectText.length)
-        // wrongStringTemplate.multiLineOffset equals zero if string in one line
-        // if wrongStringTemplate.multiLineOffset isn't equal to 0, it means that string is multiline, which means that split should be without quote
-        val isSplitInWhiteSpace = wrongStringTemplate.multiLineOffset == 0 && wrongStringTemplate.node.psi.findElementAt(wrongStringTemplate.delimiterIndex)!!.node.isWhiteSpace()
-        val correctNode =
-                if (!isSplitInWhiteSpace) {
-                    KotlinParser().createNode("$firstPart\" +\n\"$secondPart")
-                } else {
-                    // case when split space end in string interpolation
-                    // "Hello ${if (true) "Alice" else "Nick"}"
-                    //                                ^
-                    //                                |
-                    //                           split space
-                    KotlinParser().createNode("$firstPart \n$secondPart")
-                }
+        val textBetwenParts =
+            if (wrongStringTemplate.isOneLineString) {
+                "\" +\n\""
+            } else {
+                "\n"
+            }
+        val correctNode = KotlinParser().createNode("$firstPart$textBetwenParts$secondPart")
         wrongStringTemplate.node.treeParent.replaceChild(wrongStringTemplate.node, correctNode)
     }
 
@@ -486,9 +481,9 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
         /**
          * @property node node
          * @property delimiterIndex index to split
-         * @property multiLineOffset only if string is multiline
+         * @property isMultiString flag is string is multiline
          */
-        class StringTemplate(val node: ASTNode, val delimiterIndex: Int, val multiLineOffset: Int) : LongLineFixableCases()
+        class StringTemplate(val node: ASTNode, val delimiterIndex: Int, val isOneLineString: Boolean) : LongLineFixableCases()
 
         class BinaryExpression(val node: ASTNode) : LongLineFixableCases()
 

--- a/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
+++ b/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
@@ -173,7 +173,13 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
         // check, that space to split is a part of text - not code
         if (node.psi.findElementAt(delimiterIndex)!!.node.isWhiteSpace())
             return LongLineFixableCases.None
-        return LongLineFixableCases.StringTemplate(node, delimiterIndex, multiLineOffset == 0)
+        // minus 2 here as we are inserting ` +` and we don't want it to exceed line length
+        val correcterDelimiter = if (multiLineOffset ==0 && leftOffset + delimiterIndex > configuration.lineLength.toInt() - 2) {
+            node.text.substring(0, delimiterIndex - 2).lastIndexOf(' ')
+        } else {
+            delimiterIndex
+        }
+        return LongLineFixableCases.StringTemplate(node, correcterDelimiter, multiLineOffset == 0)
     }
 
     private fun checkFun(wrongNode: ASTNode) =

--- a/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
+++ b/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
@@ -278,7 +278,7 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
         node.replaceChild(whiteSpaceAfterPlus, PsiWhiteSpaceImpl("\n"))
     }
 
-    @Suppress("UnsafeCallOnNullableType")
+    @Suppress("UnsafeCallOnNullableType", "COMMENT_WHITE_SPACE")
     private fun fixStringTemplate(wrongStringTemplate: LongLineFixableCases.StringTemplate) {
         val incorrectText = wrongStringTemplate.node.text
         val firstPart = incorrectText.substring(0, wrongStringTemplate.delimiterIndex)

--- a/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
+++ b/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
@@ -6,6 +6,7 @@ import org.cqfn.diktat.common.config.rules.getRuleConfig
 import org.cqfn.diktat.ruleset.constants.Warnings.LONG_LINE
 import org.cqfn.diktat.ruleset.rules.DiktatRule
 import org.cqfn.diktat.ruleset.utils.KotlinParser
+import org.cqfn.diktat.ruleset.utils.KotlinParseException
 import org.cqfn.diktat.ruleset.utils.appendNewlineMergingWhiteSpace
 import org.cqfn.diktat.ruleset.utils.calculateLineColByOffset
 import org.cqfn.diktat.ruleset.utils.findParentNodeWithSpecificType
@@ -282,7 +283,22 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
         val incorrectText = wrongStringTemplate.node.text
         val firstPart = incorrectText.substring(0, wrongStringTemplate.delimiterIndex)
         val secondPart = incorrectText.substring(wrongStringTemplate.delimiterIndex, incorrectText.length)
-        wrongStringTemplate.node.treeParent.replaceChild(wrongStringTemplate.node, KotlinParser().createNode("$firstPart\" +\n\"$secondPart"))
+        val correctNode =
+            try {
+                KotlinParser().createNode("$firstPart\" +\n\"$secondPart")
+            } catch (ex: KotlinParseException) {
+                try {
+                    // case when split space end in string interpolation
+                    // "Hello ${if (true) "Alice" else "Nick"}"
+                    //                                ^
+                    //                                |
+                    //                           split space
+                    KotlinParser().createNode("$firstPart \n$secondPart")
+                } catch (ex: KotlinParseException) {
+                    return
+                }
+            }
+        wrongStringTemplate.node.treeParent.replaceChild(wrongStringTemplate.node, correctNode)
     }
 
     /**

--- a/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
+++ b/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
@@ -295,6 +295,7 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
         val firstPart = incorrectText.substring(0, wrongStringTemplate.delimiterIndex)
         val secondPart = incorrectText.substring(wrongStringTemplate.delimiterIndex, incorrectText.length)
         // wrongStringTemplate.multiLineOffset equals zero if string in one line
+        // if wrongStringTemplate.multiLineOffset isn't equal to 0, it means that string is multiline, which means that split should be without quote
         val isSplitInWhiteSpace = wrongStringTemplate.multiLineOffset == 0 && wrongStringTemplate.node.psi.findElementAt(wrongStringTemplate.delimiterIndex)!!.node.isWhiteSpace()
         val correctNode =
                 if (!isSplitInWhiteSpace) {
@@ -482,6 +483,11 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
             val hasNewLineBefore: Boolean,
             val indexLastSpace: Int = 0) : LongLineFixableCases()
 
+        /**
+         * @property node node
+         * @property delimiterIndex index to split
+         * @property multiLineOffset only if string is multiline
+         */
         class StringTemplate(val node: ASTNode, val delimiterIndex: Int, val multiLineOffset: Int) : LongLineFixableCases()
 
         class BinaryExpression(val node: ASTNode) : LongLineFixableCases()

--- a/diktat-rules/src/test/resources/test/paragraph3/long_line/LongStringTemplateExpected.kt
+++ b/diktat-rules/src/test/resources/test/paragraph3/long_line/LongStringTemplateExpected.kt
@@ -15,15 +15,22 @@ val someCode = 15
 class Foo() {
 
     fun Fuu() {
-        logger.log("<-- ${response.code} ${ if (response.message.isEmpty()) "skfnvkdjdfvd" else "dfjvndkjnbvif" + 
- response.message}")
-        logger.log("<-- ${response.code} ${ if (response.message.isEmpty()) "skfnvsdcsdcscskdjdfvd" else 
- "dfjvndsdcsdcsdcskjnbvif" + response.message}")
+        logger.log("<-- ${response.code} ${ if (response.message.isEmpty()) "skfnvkdjdfvd" else "dfjvndkjnbvif" + response.message}")
+        logger.log("<-- ${response.code} ${ if (response.message.isEmpty()) "skfnvsdcsdcscskdjdfvd" else "dfjvndsdcsdcsdcskjnbvif" + response.message}")
     }
 
     val q = """
-        <--${respodcnse.ccode}${if (response.mesdscsage.isEmpty()) "skfnvkdjdsdcfvd" else "dfjvndksdcjnbvif" + 
-response.mecsssdcage}
+        <--${respodcnsee.ccode}${if (response.mesdscsage.isEmpty()) "skfnvkdjeeeeeee" else "dfjvndksdcjnbvif" + response.mecsssdcage}
     """.trimIndent()
 
+    val w = """
+        first line
+    veryy looooooooooooooooong second line didfuhyg djfghdf gjh erughdjf gbdpfughb ergen r fgdngjkdg e g s g d g
+ d bd jk;g,e,r gd bnsfg e
+    """.trimIndent()
+
+    val e = """
+        another line
+        <--${respodcnse.ccode}${if (response.mesdscsage.isEmpty()) "skfnvkdjdsdcfvd" else "dfjvndksdcjnbvif" + response.mecsssdcage}
+    """.trimIndent()
 }

--- a/diktat-rules/src/test/resources/test/paragraph3/long_line/LongStringTemplateExpected.kt
+++ b/diktat-rules/src/test/resources/test/paragraph3/long_line/LongStringTemplateExpected.kt
@@ -33,4 +33,14 @@ class Foo() {
         another line
         <--${respodcnse.ccode}${if (response.mesdscsage.isEmpty()) "skfnvkdjdsdcfvd" else "dfjvndksdcjnbvif" + response.mecsssdcage}
     """.trimIndent()
+
+    fun foo() {
+        val q = """
+        re
+        ${
+            if (( x >  "436587365486374568343658736548637456834365873654863745683436587365486374568343658736548637456834365873654863745683") ||
+ y<238479283749238 && !isFoo()){}
+        }
+    """.trimIndent()
+    }
 }

--- a/diktat-rules/src/test/resources/test/paragraph3/long_line/LongStringTemplateExpected.kt
+++ b/diktat-rules/src/test/resources/test/paragraph3/long_line/LongStringTemplateExpected.kt
@@ -11,3 +11,14 @@ val someCode = 15
     @SchedulerSupport(SchedulerSupport.NONE)
     inline fun <T1 : Any, T2 : Any, T3 : Any, T4 : Any, R : Any> combineLatest() {}
 }
+
+class Foo() {
+
+    fun Fuu() {
+        logger.log("<-- ${response.code} ${ if (response.message.isEmpty()) "skfnvkdjdfvd" else "dfjvndkjnbvif" + 
+ response.message}")
+        logger.log("<-- ${response.code} ${ if (response.message.isEmpty()) "skfnvsdcsdcscskdjdfvd" else 
+ "dfjvndsdcsdcsdcskjnbvif" + response.message}")
+    }
+
+}

--- a/diktat-rules/src/test/resources/test/paragraph3/long_line/LongStringTemplateExpected.kt
+++ b/diktat-rules/src/test/resources/test/paragraph3/long_line/LongStringTemplateExpected.kt
@@ -21,4 +21,9 @@ class Foo() {
  "dfjvndsdcsdcsdcskjnbvif" + response.message}")
     }
 
+    val q = """
+        <--${respodcnse.ccode}${if (response.mesdscsage.isEmpty()) "skfnvkdjdsdcfvd" else "dfjvndksdcjnbvif" + 
+response.mecsssdcage}
+    """.trimIndent()
+
 }

--- a/diktat-rules/src/test/resources/test/paragraph3/long_line/LongStringTemplateTest.kt
+++ b/diktat-rules/src/test/resources/test/paragraph3/long_line/LongStringTemplateTest.kt
@@ -8,3 +8,12 @@ object Observables {
     @SchedulerSupport(SchedulerSupport.NONE)
     inline fun <T1 : Any, T2 : Any, T3 : Any, T4 : Any, R : Any> combineLatest() {}
 }
+
+class Foo() {
+
+    fun Fuu() {
+        logger.log("<-- ${response.code} ${ if (response.message.isEmpty()) "skfnvkdjdfvd" else "dfjvndkjnbvif" + response.message}")
+        logger.log("<-- ${response.code} ${ if (response.message.isEmpty()) "skfnvsdcsdcscskdjdfvd" else "dfjvndsdcsdcsdcskjnbvif" + response.message}")
+    }
+
+}

--- a/diktat-rules/src/test/resources/test/paragraph3/long_line/LongStringTemplateTest.kt
+++ b/diktat-rules/src/test/resources/test/paragraph3/long_line/LongStringTemplateTest.kt
@@ -29,4 +29,13 @@ class Foo() {
         another line
         <--${respodcnse.ccode}${if (response.mesdscsage.isEmpty()) "skfnvkdjdsdcfvd" else "dfjvndksdcjnbvif" + response.mecsssdcage}
     """.trimIndent()
+
+    fun foo() {
+        val q = """
+        re
+        ${
+            if (( x >  "436587365486374568343658736548637456834365873654863745683436587365486374568343658736548637456834365873654863745683") || y<238479283749238 && !isFoo()){}
+        }
+    """.trimIndent()
+    }
 }

--- a/diktat-rules/src/test/resources/test/paragraph3/long_line/LongStringTemplateTest.kt
+++ b/diktat-rules/src/test/resources/test/paragraph3/long_line/LongStringTemplateTest.kt
@@ -16,4 +16,9 @@ class Foo() {
         logger.log("<-- ${response.code} ${ if (response.message.isEmpty()) "skfnvsdcsdcscskdjdfvd" else "dfjvndsdcsdcsdcskjnbvif" + response.message}")
     }
 
+    val q = """
+        <--${respodcnse.ccode}${if (response.mesdscsage.isEmpty()) "skfnvkdjdsdcfvd" else "dfjvndksdcjnbvif" + 
+response.mecsssdcage}
+    """.trimIndent()
+
 }

--- a/diktat-rules/src/test/resources/test/paragraph3/long_line/LongStringTemplateTest.kt
+++ b/diktat-rules/src/test/resources/test/paragraph3/long_line/LongStringTemplateTest.kt
@@ -17,8 +17,16 @@ class Foo() {
     }
 
     val q = """
-        <--${respodcnse.ccode}${if (response.mesdscsage.isEmpty()) "skfnvkdjdsdcfvd" else "dfjvndksdcjnbvif" + 
-response.mecsssdcage}
+        <--${respodcnsee.ccode}${if (response.mesdscsage.isEmpty()) "skfnvkdjeeeeeee" else "dfjvndksdcjnbvif" + response.mecsssdcage}
     """.trimIndent()
 
+    val w = """
+        first line
+    veryy looooooooooooooooong second line didfuhyg djfghdf gjh erughdjf gbdpfughb ergen r fgdngjkdg e g s g d g d bd jk;g,e,r gd bnsfg e
+    """.trimIndent()
+
+    val e = """
+        another line
+        <--${respodcnse.ccode}${if (response.mesdscsage.isEmpty()) "skfnvkdjdsdcfvd" else "dfjvndksdcjnbvif" + response.mecsssdcage}
+    """.trimIndent()
 }

